### PR TITLE
Grammatical Tweaks to the Docs

### PR DESCRIPTION
**Pull request by ControlCplusControlV**
I just went through the docs with Grammarly trying to find any issues in the docs, it picked up some issues, while other changes can be viewed as more optional (sometimes redundancy is necessary when trying illustrate a point) so remove or tweak back anything you want to stay the same, and if not hopefully this caught atleast a couple things that should be changed, I also tried to stay within common sense and reason and not force any particular dialect of English onto the Docs!

### CODE REVIEW

File: docs/src/advanced/traits.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## Declaring a Trait
 
-A _trait_ opts a type into a certain type of behavior or functionality that can be shared among types. This allows for easy reuse of code and generic programming. If you have ever used a typeclass in Haskell, a trait in Rust, or even an interface in Java, these are similar concepts.
+A _trait_ opts a type into a certain type of behavior or functionality that can be shared among types. This allows for easy reuse of code and generic programming. If you have ever used a type class in Haskell, a trait in Rust, or even an interface in Java, these are similar concepts.
```
Comment (adlerjohn) [inline docs/src/advanced/traits.md:5]:
```suggestion
A _trait_ opts a type into a certain type of behavior or functionality that can be shared among types. This allows for easy reuse of code and generic programming. If you have ever used a typeclass in Haskell, a trait in Rust, or even an interface in Java, these are similar concepts.
```

File: docs/src/basics/built_in_types.md [hunk 2]
```diff
@@ -16,13 +16,13 @@ Sway has the following primitive types:
 1. `bool` (Boolean `true` or `false`)
 1. `b256` (256 bits (32 bytes), i.e. a hash)
 
-All other types in Sway are built up of these primitive types, or references to these primitive types. You may notice that there are no signed integers&mdash;this is by design. In the blockchain domain that Sway occupies, floating point values and negative numbers have smaller utility, so their implementation has been left up to libraries for specific use cases.
+All other types in Sway are built up of these primitive types, or references to these primitive types. You may notice that there are no signed integers&mdash; this is by design. In the blockchain domain that Sway occupies, floating-point values and negative numbers have smaller utility, so their implementation has been left up to libraries for specific use cases.
```
Comment (adlerjohn) [inline docs/src/basics/built_in_types.md:19]: I guess Grammarly doesn't know html 😂

```suggestion
All other types in Sway are built up of these primitive types, or references to these primitive types. You may notice that there are no signed integers&mdash;this is by design. In the blockchain domain that Sway occupies, floating-point values and negative numbers have smaller utility, so their implementation has been left up to libraries for specific use cases.
```

Comment (adlerjohn) [inline docs/src/basics/built_in_types.md:19]: I guess Grammarly doesn't know html 😂

```suggestion
All other types in Sway are built up of these primitive types, or references to these primitive types. You may notice that there are no signed integers&mdash;this is by design. In the blockchain domain that Sway occupies, floating-point values and negative numbers have smaller utility, so their implementation has been left up to libraries for specific use cases.
```

File: docs/src/sway-on-chain/smart_contracts.md [hunk 3]
```diff
@@ -43,12 +43,12 @@ In the second line,
     fn receive_funds(gas: u64, coins_to_forward: u64, asset_id: b256, unused: ());
 ```
 
-we are declaring an ABI interface surface method called `receive funds` which, when called, should receive funds into this wallet. Note that we are simply defining an interface here, so there is no _function body_, or implementation of the function. We only need to define the interface itself. In this way, ABI declarations are similar to [trait declarations](../advanced/traits.md). This ABI method takes four parameters: `gas`, `coins_to_forward`, `asset_id`, and `unused`, and doesn't return anything.
+we are declaring an ABI interface surface method called `receive funds` which, when called, should receive funds into this wallet. Note that we are simply defining an interface here, so there is no _function body_or implementation of the function. We only need to define the interface itself. In this way, ABI declarations are similar to [trait declarations](../advanced/traits.md). This ABI method takes four parameters: `gas`, `coins_to_forward`, `asset_id`, and `unused`, and doesn't return anything.
```
Comment (adlerjohn) [inline docs/src/sway-on-chain/smart_contracts.md:46]:
```suggestion
we are declaring an ABI interface surface method called `receive funds` which, when called, should receive funds into this wallet. Note that we are simply defining an interface here, so there is no _function body_ or implementation of the function. We only need to define the interface itself. In this way, ABI declarations are similar to [trait declarations](../advanced/traits.md). This ABI method takes four parameters: `gas`, `coins_to_forward`, `asset_id`, and `unused`, and doesn't return anything.
```

**Review by adlerjohn — CHANGES_REQUESTED** [review]

**Comment by ControlCplusControlV** [comment]
@adlerjohn I added your changes I am pretty sure, I think I did it in the wrong order though so some got pushed to outdated and can't be directly commited, but combing through it looks like they were added. Lmk though if I missed something and another change is needed

**Review by adlerjohn — APPROVED** [review]

**Review by adlerjohn — APPROVED** [review]

**Pull request merged**